deepmd/loss/ener.py

```diff
@@ -365,151 +363,202 @@ def build(self, learning_rate, natoms, model_dict, label_dict, suffix):
+        force_m_pred = paddle.reshape(
+            paddle.slice(
+                force_pred_reshape,
+                [0, 1],
+                [0, real_nloc * 3],
+                [force_pred_reshape.shape[0], real_nloc * 3 + virt_nloc * 3],
+            ),
+            [-1],
+        )
+        force_r_label = paddle.reshape(
+            paddle.slice(
+                force_label_reshape,
+                [0, 1],
+                [0, 0],
+                [force_label_reshape.shape[0], real_nloc * 3],
+            ),
+            [-1],
+        )
+        force_m_label = paddle.reshape(
+            paddle.slice(
+                force_label_reshape,
+                [0, 1],
+                [0, real_nloc * 3],
+                [force_label_reshape.shape[0], real_nloc * 3 + virt_nloc * 3],
+            ),
+            [-1],
         )
-        l2_force_r_loss = tf.reduce_mean(
-            tf.square(force_r_pred - force_r_label), name="l2_force_real_" + suffix
+        l2_force_r_loss = (
+            paddle.mean(
+                paddle.square(force_r_pred - force_r_label),
+                name="l2_force_real_" + suffix,
+            )
+            if self.has_fr
+            else 0.0
         )
-        l2_force_m_loss = tf.reduce_mean(
-            tf.square(force_m_pred - force_m_label), name="l2_force_mag_" + suffix
+        l2_force_m_loss = (
+            paddle.mean(
+                paddle.square(force_m_pred - force_m_label),
+                name="l2_force_mag_" + suffix,
+            )
+            if self.has_fm
+            else 0.0
         )
 
-        virial_pred_reshape = tf.reshape(virial_pred, [-1])
-        virial_label_reshape = tf.reshape(virial_label, [-1])
-        l2_virial_loss = tf.reduce_mean(
-            tf.square(virial_pred_reshape - virial_label_reshape),
-            name="l2_virial_" + suffix,
+        virial_pred_reshape = paddle.reshape(virial_pred, [-1])
+        virial_label_reshape = paddle.reshape(virial_label, [-1])
+        l2_virial_loss = (
+            paddle.mean(
+                paddle.square(virial_pred_reshape - virial_label_reshape),
+                name="l2_virial_" + suffix,
+            )
+            if self.has_v
+            else 0.0
         )
 
         # need to change. can't get atom_ener_hat?
-        atom_ener_pred_reshape = tf.reshape(atom_ener_pred, [-1])
-        atom_ener_label_reshape = tf.reshape(atom_ener_label, [-1])
-        l2_atom_ener_loss = tf.reduce_mean(
-            tf.square(atom_ener_pred_reshape - atom_ener_label_reshape),
-            name="l2_atom_ener_" + suffix,
+        atom_ener_pred_reshape = paddle.reshape(atom_ener_pred, [-1])
+        atom_ener_label_reshape = paddle.reshape(atom_ener_label, [-1])
+        l2_atom_ener_loss = (
+            paddle.mean(
+                paddle.square(atom_ener_pred_reshape - atom_ener_label_reshape),
+                name="l2_atom_ener_" + suffix,
+            )
+            if self.has_ae
+            else 0.9
```

Comment:
I guess it is a typo..
```suggestion
            else 0.0
```

Reply:
Fixed, thanks!